### PR TITLE
Fixes selective tests in case of missing merge commits

### PR DESCRIPTION
**Pull request by potiuk**
In case of very simple changes, there might be no merge commits
generated by GitHub. In such cases we should take the commit SHA
instead as the base of change calculation for selective tests.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
¯\_(ツ)_/¯  - some of Github inconsistencies. Any of the committers - some PRs might be failing until this one is merged https://github.com/apache/airflow/pull/11641 - seems that GH merge commit vs. lack of it is rather interesting story (works sometimes). 
The PR should work fine in both cases and fall-back to full tests in case there are some troubles)

**Comment by potiuk** [comment]
Would love if someone approves/merges this one if I fall asleep in the meantime :)

**Review by kaxil — APPROVED** [review]
:/

**Pull request merged**
